go/multiorch/coordinator/leader_appointment.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coordinator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/multigres/multigres/go/mterrors"
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+	mtrpcpb "github.com/multigres/multigres/go/pb/mtrpc"
+	multipoolermanagerdatapb "github.com/multigres/multigres/go/pb/multipoolermanagerdata"
+)
+
+// BeginTerm implements stages 1-4 of the consensus protocol:
+// 1. Discover max term from all nodes
+// 2. Increment to get new term
+// 3. Select candidate based on WAL position
+// 4. Send BeginTerm RPC to all nodes in parallel
+// 5. Validate quorum (majority acceptance)
+//
+// Returns the candidate node, standbys, the new term, and any error.
+func (c *Coordinator) BeginTerm(ctx context.Context, shardID string, cohort []*Node) (*Node, []*Node, int64, error) {
+	// Stage 1: Obtain Term Number - Query all nodes for max term
+	c.logger.InfoContext(ctx, "Discovering max term", "shard", shardID, "cohort_size", len(cohort))
+	maxTerm, err := c.discoverMaxTerm(ctx, cohort)
```

Comment:
Idea here:
- Coordinator gets initialized with a MaxTerm on process start. That should be sufficient.
- As part of maintenance tasks in operators, we are periodically checking the current max term. Only reason why that would diverge is that an election happen that this coordinator hasn't discovered yet. I think that could be part of maintance tick instead of trying to do it at the moment of the election.